eth/tx_forwader.go

```diff
@@ -0,0 +1,63 @@
+package eth
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const (
+	// defaultDialTimeout is default duration the service will wait on
+	// startup to make a connection to either the L1 or L2 backends.
+	defaultDialTimeout = 5 * time.Second
+)
+
+// dialSequencerClientWithTimeout attempts to dial the Sequencer using the
+// provided URL. If the dial doesn't complete within defaultDialTimeout
+// seconds, this method will return an error.
+func dialSequencerClientWithTimeout(ctx context.Context, url string) (
+	*ethclient.Client, error) {
+
+	ctxt, cancel := context.WithTimeout(ctx, defaultDialTimeout)
+	defer cancel()
+
+	return ethclient.DialContext(ctxt, url)
+}
+
+// TxForwarder connects to a sequencer http endpoint and forwards mempool
+// transactions to it.
+type TxForwarder struct {
+	SequencerHTTP string
+	Client        *ethclient.Client
+	ctx           context.Context
+}
+
+// Dial connects to the sequencer http endpoint. It only needs to be called
+// once after initializing a new forwarder.
+func (f *TxForwarder) Dial() error {
+	client, err := dialSequencerClientWithTimeout(f.ctx, f.SequencerHTTP)
+	if err != nil {
+		return err
+	}
+	f.Client = client
+	return nil
+}
+
+// Forward sends the transaction to the sequencer.
+func (f *TxForwarder) Forward(tx types.Transaction) error {
+	if f.Client == nil {
+		return errors.New("cannot forward without Dial")
+	}
+	return nil
```

Comment:
It's not forwarding yet? Should there be an ethclient call here?

Reply:
Yeah, sorry missed that.